src/Microsoft.AspNet.Razor.Runtime/TagHelpers/DefaultTagHelperContent.cs

```diff
@@ -158,66 +141,119 @@ public override TagHelperContent AppendFormat([NotNull] string format, params ob
             [NotNull] string format,
             params object[] args)
         {
-            _buffer.Add(string.Format(provider, format, args));
+            Buffer.Append(string.Format(provider, format, args));
             return this;
         }
 
         /// <inheritdoc />
-        public override TagHelperContent Append(TagHelperContent tagHelperContent)
+        public override TagHelperContent Append(IHtmlContent htmlContent)
         {
-            if (tagHelperContent != null)
-            {
-                foreach (var value in tagHelperContent)
-                {
-                    Append(value);
-                }
-            }
-
-            // If Append() was called with an empty TagHelperContent IsModified should
-            // still be true. If the content was not already modified, it means it is empty.
-            // So the Clear() method can be used to indirectly set the IsModified.
-            if (!IsModified)
-            {
-                Clear();
-            }
-
+            Buffer.Append(htmlContent);
             return this;
         }
 
         /// <inheritdoc />
         public override TagHelperContent Clear()
         {
-            _buffer.Clear();
+            Buffer.Clear();
             return this;
         }
 
         /// <inheritdoc />
         public override string GetContent()
         {
-            return string.Join(string.Empty, _buffer);
+            if (_buffer == null)
+            {
+                return string.Empty;
+            }
+
+            return string.Join(string.Empty, Buffer);
         }
 
         /// <inheritdoc />
         public override string ToString()
         {
-            return GetContent();
+            using (var writer = new StringWriter())
+            {
+                WriteTo(writer, new HtmlEncoder());
```

Comment:
If we can remove this method that'd be best. We already have a way to return the raw version and encoded version ofthe string. Don't want to make `ToString` magic.